test/e2e/framework/gpu_util.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+)
+
+const (
+	// GPUResourceName is the extended name of the GPU resource since v1.8
+	// this uses the device plugin mechanism
+	NVIDIAGPUResourceName = "nvidia.com/gpu"
+
+	// GPUDevicePluginDSYAML is the official Google Device Plugin Daemonset NVIDIA GPU manifest for GKE
+	GPUDevicePluginDSYAML = "https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/master/device-plugin-daemonset.yaml"
+)
+
+// NumberOfGPUs returs the number of GPUs advertised by a node
+// This is based on the Device Plugin system and expected to run on a COS based node
+// After the NVIDIA drivers were installed
+func NumberOfNVIDIAGPUs(node *v1.Node) int64 {
+	val, ok := node.Status.Capacity[NVIDIAGPUResourceName]
+
+	if !ok {
+		return 0
+	}
+
+	return val.Value()
+}
+
+// NVIDIADevicePlugin returns the official Google Device Plugin pod for NVIDIA GPU in GKE
+func NVIDIADevicePlugin(ns string) *v1.Pod {
```

Comment:
I would insist that we don't put these definitions in the framework package.

Both the methods: `NumberOfNVIDIAGPUs()` and `NVIDIADevicePlugin()` are used only in `e2e_node/gpu-device-plugin.go` and can live there.

Reply:
`NumberOfNVIDIAGPUs()` is only used in `e2e_node/gpu-device-plugin.go` mostly because I don't want to make use of it in e2e/scheduling/nvidia-gpu` in this PR but in a later PR.

To answer your other concern, I do believe that we need a utils file that groups the utils functions used in the NVIDIA tests as they won't be used in one single package across the e2e code base.

The package `framework` seems to be doing exactly that (regroup all the `utils` function) which is why it seems to me that it's the right place.

@jiayingz mentioned that:
> Our main concern is these definitions are gpu specific and irrelevant to most people using e2e framework.

But that argument can be applied to a lot of utils functions the framework package is exposing (see pv_utils, rc_utils, nodes_utils, perf_utils, utils, exec_utils, statefulset_utils, firewall_utils).
Granted that some make more sense than others (API utils vs platform specific utils) but gpu_utils doesn't feel completely out of place when looking at a lot of these utils functions.